proxy/logging/LogField.cc

```diff
@@ -218,14 +218,19 @@ LogField::init_milestone_container()
 }
 
 // Generic field ctor
-LogField::LogField(const char *name, const char *symbol, Type type, MarshalFunc marshal, UnmarshalFunc unmarshal, SetFunc _setfunc)
+LogField::LogField(const char *name, const char *symbol, Type type, MarshalFunc marshal, VarUnmarshalFuncSliceOnly unmarshal,
+                   SetFunc _setfunc)
   : m_name(ats_strdup(name)),
     m_symbol(ats_strdup(symbol)),
     m_type(type),
     m_container(NO_CONTAINER),
     m_marshal_func(marshal),
-    m_unmarshal_func(unmarshal),
-    m_unmarshal_func_map(nullptr),
+    m_unmarshal_func([](VarUnmarshalFuncSliceOnly f) -> VarUnmarshalFunc {
```

Comment:
Do you want to pass `f` by value?

Reply:
Hmm I guess I'm assuming it would optimize the same as a const ref?  The size will be the same as `struct { char t; void *p };` in any case, pretty sure.